src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -102,16 +117,59 @@ public void ValidateConfig()
             // Connection string can't be null or empty
             if (string.IsNullOrWhiteSpace(dataSource.ConnectionString))
             {
-                throw new DataApiBuilderException(
-                    message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
-                    statusCode: HttpStatusCode.ServiceUnavailable,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                HandleOrRecordException(
+                    new DataApiBuilderException(
+                        message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization)
+                );
             }
         }
 
         ValidateDatabaseType(runtimeConfig, fileSystem, logger);
     }
 
+    public async void ValidateConfig(string configFilePath)
+    {
+        _isValidateOnly = true;
+        await ValidateConfigSchema(configFilePath);
+        ValidateConfigProperties();
+        ValidatePermissionsInConfig(_runtimeConfigProvider.GetConfig());
+        await ValidateEntitiesMetadata();
+        _isValidateOnly = false;
+    }
+
+    public async Task ValidateConfigSchema(string configFilePath)
+    {
+        string jsonData = _fileSystem.File.ReadAllText(configFilePath);
+        await JsonConfigSchemaValidator.ValidateJsonConfig(jsonData);
+    }
+
+    public async Task ValidateEntitiesMetadata()
+    {
+        LoggerFactory loggerFactory = new();
+        QueryManagerFactory queryManagerFactory = new(
+            runtimeConfigProvider: _runtimeConfigProvider,
+            logger: loggerFactory.CreateLogger<IQueryExecutor>(),
+            contextAccessor: null!);
+        
+        MetadataProviderFactory metadataProviderFactory = new(
```

Comment:
validation when triggered by cli, won't have these setup in StartUp, because it's role is simply to validate the config file. It won't setup the application

Reply:
In that case, we should create these objects only in the CLI validate implementation. Not create duplicates for this in the Service. Do we perform the ValidateRelationshipsInConfig in the service too?

If these are new objects we are adding as dependencies to the RuntimeConfigValidator, we should do that. And in the cli, create a RuntimeConfigValidator and supply all these objects explicitly.

Reply:
I think we should keep the existing `RuntimeConfigValidator` untouched and all the additional work that dab validate needs to perform goes into another class which will have these extra added dependencies.

Reply:
I'm trying to have a single method in RuntimeConfigValidator class which is called by CLI for validate